minimint/tests/fixture/mod.rs

```diff
@@ -0,0 +1,594 @@
+use minimint_api::db::batch::DbBatch;
+use minimint_api::db::mem_impl::MemDatabase;
+use minimint_api::db::Database;
+use minimint_api::{Amount, FederationModule, OutPoint, PeerId};
+use minimint_ln::contracts::Contract;
+use minimint_ln::{ContractOrOfferOutput, ContractOutput};
+use minimint_mint::PartiallySignedRequest;
+use minimint_wallet::bitcoind::BitcoindRpc;
+use minimint_wallet::config::WalletConfig;
+use minimint_wallet::db::UnsignedTransactionKey;
+use minimint_wallet::txoproof::TxOutProof;
+use minimint_wallet::{Wallet, WalletConsensusItem};
+use mint_client::api::HttpFederationApi;
+use mint_client::clients::gateway::{GatewayClient, GatewayClientConfig};
+use mint_client::ln::gateway::LightningGateway;
+use mint_client::UserClient;
+use real::{RealBitcoinTest, RealLightningTest};
+
+mod fake;
+mod real;
+
+static BASE_PORT: AtomicU16 = AtomicU16::new(4000 as u16);
+
+// Helper functions for easier test writing
+pub fn rng() -> OsRng {
+    OsRng::new().unwrap()
+}
+
+pub fn sats(amount: u64) -> Amount {
+    Amount::from_sat(amount)
+}
+
+/// Generates the fixtures for an integration test and spawns API and HBBFT consensus threads for
+/// federation nodes starting at port 4000.
+pub async fn fixtures(
+    num_peers: u16,
+    max_evil_threshold: usize,
+    amount_tiers: &[Amount],
+) -> (
+    FederationTest,
+    UserTest,
+    Box<dyn BitcoinTest>,
+    GatewayTest,
+    Box<dyn LightningTest>,
+) {
+    let base_port = BASE_PORT.fetch_add(num_peers * 2, Ordering::Relaxed);
+
+    // in case we need to output logs using 'cargo test -- --nocapture'
+    if base_port == 4000 {
+        tracing_subscriber::fmt()
+            .with_env_filter(
+                EnvFilter::try_from_default_env()
+                    .unwrap_or_else(|_| EnvFilter::new("info,tide=error")),
+            )
+            .init();
+    }
+
+    let params = ServerConfigParams {
+        hbbft_base_port: base_port,
+        api_base_port: base_port + num_peers,
+        amount_tiers: amount_tiers.to_vec(),
+    };
+    let peers = (0..num_peers as u16).map(PeerId::from).collect::<Vec<_>>();
+
+    let (server_config, client_config) = ServerConfig::trusted_dealer_gen(
+        &peers,
+        max_evil_threshold,
+        &params,
+        OsRng::new().unwrap(),
+    );
+
+    match env::var("MINIMINT_TEST_REAL") {
+        Ok(s) if s == "1" => {
+            info!("Testing with REAL Bitcoin and Lightning services");
+            let dir =
+                env::var("MINIMINT_TEST_DIR").expect("Must have test dir defined for real tests");
+            let wallet_config = server_config.iter().last().unwrap().1.wallet.clone();
+            let bitcoin_rpc = Wallet::bitcoind(wallet_config.clone());
+            let bitcoin = RealBitcoinTest::new(wallet_config);
+            let socket_gateway = PathBuf::from(dir.clone() + "ln1/regtest/lightning-rpc");
+            let socket_other = PathBuf::from(dir + "ln2/regtest/lightning-rpc");
```

Comment:
Use `PathBuf::from(dir).join("ln1/regtest/lightning-rpc")` instead. Using `+` depends on having a trailing slash in `MINIMINT_TEST_REAL`.

Reply:
Good to know, I'll change that!